packages/mask/src/plugins/Omen/UI/OmenDetailsCard.tsx

```diff
@@ -0,0 +1,121 @@
+import BigNumber from 'bignumber.js'
+import { makeStyles } from '@masknet/theme'
+import { Card, CardContent, Typography, Grid, Box } from '@mui/material'
+import { useFetchToken } from '../hooks/useToken'
+import { useI18N } from '../../../utils/i18n-next-ui'
+import type { fpmmLiquidity, tokenData } from '../types'
+
+const useStyles = makeStyles()((theme) => ({
+    root: {
+        marginTop: theme.spacing(2),
+    },
+    card: {
+        width: '100%',
+        boxShadow: 'none',
+        border: `solid 1px ${theme.palette.divider}`,
+        alignItems: 'center',
+    },
+    gridRow: {
+        width: '100%',
+        marginLeft: theme.spacing(2),
+    },
+    gridCol: {
+        margin: theme.spacing(1),
+    },
+    gridItemTitle: {
+        margin: '3px',
+        fontSize: 12,
+    },
+    gridItemValue: {
+        fontSize: 12,
+    },
+}))
+
+interface OmenDetailsCardProps {
+    liquidityData: fpmmLiquidity[]
+    tokenId: string
+    usdRunningDailyVolume: number
+    closingTimestamp: Date
+    collateralVolume: BigNumber
+}
+
+export function OmenDetailsCard(props: OmenDetailsCardProps) {
+    const { t } = useI18N()
+    const { classes } = useStyles()
+    const { liquidityData, tokenId, usdRunningDailyVolume, closingTimestamp, collateralVolume } = props
+    const tokenDataRes = useFetchToken(tokenId)
+    const tokenObj: tokenData | null | undefined = tokenDataRes ? tokenDataRes.value : undefined
+    const collateralToken = tokenObj?.registeredToken
+    const collateralDecimals = collateralToken ? collateralToken.decimals : 0
+    let sharesAmount = new BigNumber(0)
+    const liquidityArr = Object.values(liquidityData)
+    for (const liquidityObj of liquidityArr) {
+        const tempSum = new BigNumber(sharesAmount)
+        const nextVal = new BigNumber(liquidityObj.sharesAmount)
+        sharesAmount = tempSum.plus(nextVal)
```

Comment:
Looks strange to me, only the last item of `liquidityArr` will take into account.